### ISSUE TITLE
[1.11.2] Rendering Error Causing Crash

### ISSUE THREAD

**Issue by bfriebe** [RootsTeam/Embers#179]
Ok so my friend has started to get more into this awesome mod on my server, and everything was fine. I restarted the server a little while ago and now when I'm near his Ember basic setup or even look towards it, our games crash. I've looked in the crash log, and it seems to change from different blocks. The first one I had seen was referencing TileEntityRender of the furnace top. The most recent references are for the Stamp Base.
First Crash Report: https://pastebin.com/YEJXeAmD
Second Crash Report: https://pastebin.com/GhCFNMaV

This mod is installed within the "The New Frontier" modpack available on Curse/Twitch app. It used to work perfectly fine.
Thanks for any and all help.

**Comment by Leviathan143** [comment]
I've been poking around in the code, this crash seems to be related to getting a fluid texture. Do you have any idea what fluid might have been in the melter and the stamp base?

**Comment by bfriebe** [comment]
There was most likely silver in the furnace, but it is possible it could
have been aluminum. My friend did copper iron and gold fine, the he did
silver or aluminum and that is when the crashing started.

**Comment by Leviathan143** [comment]
I'm currently trying to reproduce this crash with a setup consisting of two stampers and two melters. One stamper and one melter contains molten aluminium, the other stamper and melter contains molten silver. 

**Setups I have tried:**  
* Client connected to integrated server. Development environment. Failed to reproduce crash.
* Client connected to dedicated server. Development environment. Failed to reproduce crash.

**Setups I plan to try:**  
* Client connected to integrated server. Non-development environment.
* Client connected to dedicated server. Non-development environment.
* Client connected to dedicated server. Non-development environment. Client spawns out of render range of test setup and then moves in range. If successful in reproducing crash, attempt to reproduce in development environment.

**Comment by bfriebe** [comment]
ok So I have a dedicated server running on a dell poweredge r710 in my basement. The server is running on Ubuntu server. When I say "our games crash" I mean that both my friend's and my games will crash when apparently trying to render it. He first crashed when he put the ore in and it melted. I crashed when I went near his base. Only the clients crash. Server is fine and keeps going. When reloading minecraft and logging back into the server, I have a few seconds to basically look away from the general direction of the ember setup, and get away. Otherwise, if even look in the direction of it (when within rendering distance), my client will crash.

**Comment by bfriebe** [comment]
Ok update. Sorry to more or less have wasted your time, but it turns out it is in fact tin, not silver or aluminum that causes the crash. I was able to reproduce the problem in single player. There are at least 4 types of tin in the modpack. TSONcraft, NeoTech, Embers, NuclearCraft. I will update this when I narrow down more specifically what is happening.

**Comment by Leviathan143** [comment]
Excellent! I'll try tin later when I have time. If I can reproduce the issue consistently, it'll be much easier to fix.

**Comment by elucent** [comment]
I can't reproduce this either. Let me know if you find a consistent way of causing the crash.

**Comment by Happynger** [comment]
This issue happens in my single player world. In it, i did notice that when iron and other metals melted in, for example, Tinker's construct smeltery there was no rendering except for common checker of pink and black. Also, most of the buckets in the jei had no texture. This did not create any crashes until it was in the ember's melter. The trouble is that when I try to recreate the issue in the creative world all of the metal's texture is back and no crashes occur. I was playing on the modpack from Curse/Twitch client called (All the Mods 2) version v0.63b (latest). The same texture missing issue didn't occur even if I create a copy of my world.
![2017-05-14_10 09 08](https://cloud.githubusercontent.com/assets/13449795/26037099/254feb1e-38a1-11e7-9d5b-bb321dfc194a.png)

![2017-05-14_11 21 11](https://cloud.githubusercontent.com/assets/13449795/26037095/0f697086-38a1-11e7-978e-369b4fcab0f1.png)

![2017-05-14_12 29 37](https://cloud.githubusercontent.com/assets/13449795/26037096/15334eba-38a1-11e7-9cc3-d84a38eb3461.png)

[crash-2017-05-14_00.17.44-client.txt](https://github.com/RootsTeam/Embers/files/999667/crash-2017-05-14_00.17.44-client.txt)

[crash-2017-05-14_00.13.46-client.txt](https://github.com/RootsTeam/Embers/files/999668/crash-2017-05-14_00.13.46-client.txt)

Hope some of this data will help as I myself is not very familiar with Minecraft debugging. GL

**Comment by elucent** [comment]
That's a 1.10.2 instance, I'm not maintaining the mod for that version anymore.

**Comment by Happynger** [comment]
As I said previously I am not good at debugging but the minecraft version is 1.11.2 also the embers mod version is 0.212 (embers-0.212.jar). I might be making a mistake in my assumptions as I am tired. Sorry for that.

**Comment by bfriebe** [comment]
So it seems that I can't reliably reproduce the crash, but my friend and I are sure it has to do with the melter. It could be the rendering of the smoke or trying to render the fluids. Like @Happynger, I too have had it were some textures in the Tinker's smeltery were not showing up and had the purple and black. This is not always the case though, it seems to come and go. I wonder if when the textures aren't appearing/rendering that is what causes the issues with the melter. This is all speculation, but I really don't know what else. I want to thank all of you for all the time you have spent looking into this. I did manage to fix it (by keeping the chunk loaded and letting all the metal process out into ingots).

**Comment by keyangel** [comment]
Same issue:  https://pastebin.com/PZKeC4z3

Mod pack is All the Mods 2 (which is a 1.11.2 version)

I suspect there might be some kind of conflict between mods creating various liquids.  In the pack, when you look up Embers in JEI it's molten metals all have errored names.

Basically, melting anything down in the Melter that doesn't have a texture is causing a crash.

**Comment by elucent** [comment]
Fluids are weird. I'll take another crack at it when I look at Embers again.

**Comment by LittleODie** [comment]
This issues persists the same as described in "The New Frontier". I had to move and start over as I was not able to even look at my embers setup without client crash. No new information to add other than it is not just on All The Mods 2.

**Comment by bfriebe** [comment]
We more or less figured it was the rendering of missing liquid textures. Oh
and if you happend to have your setup auto make ingots if you load the
chunk it will process out and you will be ok.

**Comment by Chromogen** [comment]
oh well... have another render crash :/

https://pastebin.com/30MSscDi

happened when i put iron in the melter and connected it to embers
engine crashed as soon as the iron wanted to become liquid

also, for some reason i noticed that molten iron texture ->in this save<- was broken. 
in another creative world everything was just fine. tested setup, worked like a charm.

maybe it broke cause i added another mod later in?
remember i added TiC later on and i m pretty sure it is the only other mod with molten iron...

**Comment by Chromogen** [comment]
alright, so i removed TiC and loaded the old savegame. and it worked!
guess it really is something with the the way of fluid registering of TiC and or the rendering of embers.
its just strage that this seems to happen if TiC is added later on...

**Comment by bfriebe** [comment]
TiC was added later on in mine Maxwell because the modpack updated and the
author decided to add it.

**Comment by sandalle** [comment]
I had the same client CTD after adding Tinker's Construct (TiC) 1.11.2-2.7.0.26 with Embers 0.214 when looking in the direction of my Embers setup. IIRC, I had Lead in the Embers smeltery. I removed TiC, and removed the Embers smeltery and stamper setup, then re-added TiC and and I could play in my world, including the Embers area (very simple setup so far) just fine.

**Comment by KizerKitty** [comment]
[2017-06-01-2.log.gz](https://github.com/RootsTeam/Embers/files/1046243/2017-06-01-2.log.gz)
Can confirm that this happens when TiC is added and an old world save is added with Embers. Though it does not CTD with Liquid dawnstone present, the other's do for some reason. This occurs with either the stamper base or the fluid tanks having the liquid metal in them and rendered in the world but not when inventory.

**Comment by sandalle** [comment]
An additional point, a new Minecraft 1.11 world created with both Embers and TiC seems to work fine, including the fluid textures. The issue occurs when adding TiC _after_ the world is created with Embers installed. I did not try the reverse, a world created with TiC first and then Embers added later.

**Comment by elucent** [comment]
I've added nullchecks so that this issue will not crash people anymore. I will leave this thread open because I believe the addition of Tinkers' Construct will simply prevent the fluids from rendering -- but at least it won't crash anymore, right?

Fluids are really a mess.

**Comment by sandalle** [comment]
Embers 0.215 now no longer crashes my pre-existing world with TiC added after the fact, thank you! :) The fluid textures in the Smeltery (e.g. iron and bronze) are still missing their textures, as you noted.

**Comment by elucent** [comment]
Fluids are just messed up. I'll look into fixing the issue when I rework bits of Embers for 1.12, but it's a deep-rooted problem.

**Comment by bfriebe** [comment]
Yes, but I don't think there is much you can do about it. Thank you so much
for putting in a fix though. As long as it works it doesn't matter (to most
of us at least) if there are missing textures for molten metals.
